lib/puma/events.rb

```diff
@@ -106,10 +106,12 @@ def parse_error(error, req)
     end
 
     # An SSL error has occurred.
-    # +error+ an exception object, +peeraddr+ peer address,
-    # and +peercert+ any peer certificate (if present).
+    # @param error <Puma::MiniSSL::SSLError>
+    # @param ssl_socket <Puma::MiniSSL::Socket>
     #
-    def ssl_error(error, peeraddr, peercert)
+    def ssl_error(error, ssl_socket)
+      addr = ssl_socket.peeraddr.last rescue "<unknown>"
+      peercert = ssl_socket.peercert
       subject = peercert ? peercert.subject : nil
       @error_logger.info(error: error, text: "SSL error, peer: #{peeraddr}, peer cert: #{subject}")
```

Comment:
peeraddr is undefined. did you mean to change this line to `peer: #{addr}`?

Reply:
You're being too nice.  Actually, I'm changing `addr` on line 113 to `peeraddr`.  More importantly, I'm adding a test (bad assumption on my part).  Give me a minute...

Reply:
Fixed and added a test.  Thanks.